lib/binding_web/src/query.ts

```diff
@@ -198,19 +211,322 @@ export class Query {
+                throw new Error(
+                  `Wrong number of arguments to \`#set!\` predicate. Expected 1 or 2. Got ${steps.length - 1}.`,
+                );
+              }
+              if (!steps.every(isStringStep)) {
+                throw new Error(
+                  `Arguments to \`#set!\` predicate must be strings.".`,
+                );
+              }
+              if (!setProperties[i]) setProperties[i] = {};
+              setProperties[i][steps[1].value] = steps[2]?.value ?? null;
+              break;
+            }
+
+            case 'is?':
+            case 'is-not?': {
+              if (steps.length < 2 || steps.length > 3) {
+                throw new Error(
+                  `Wrong number of arguments to \`#${operator}\` predicate. Expected 1 or 2. Got ${steps.length - 1}.`,
+                );
+              }
+              if (!steps.every(isStringStep)) {
+                throw new Error(
+                  `Arguments to \`#${operator}\` predicate must be strings.".`,
+                );
+              }
+              const properties = operator === 'is?' ? assertedProperties : refutedProperties;
+              if (!properties[i]) properties[i] = {};
+              properties[i][steps[1].value] = steps[2]?.value ?? null;
+              break;
+            }
+
+            case 'not-any-of?':
+              isPositive = false;
+            case 'any-of?': {
+              if (steps.length < 2) {
+                throw new Error(
+                  `Wrong number of arguments to \`#${operator}\` predicate. Expected at least 1. Got ${steps.length - 1}.`,
+                );
+              }
+              if (steps[1].type !== 'capture') {
+                throw new Error(
+                  `First argument of \`#${operator}\` predicate must be a capture. Got "${steps[1].value}".`,
+                );
+              }
+              captureName = steps[1].name;
+
+              const stringSteps = steps.slice(2);
+              if (!stringSteps.every(isStringStep)) {
+                throw new Error(
+                  `Arguments to \`#${operator}\` predicate must be strings.".`,
+                );
+              }
+              const values = stringSteps.map((s) => s.value);
+
+              textPredicates[i].push((captures) => {
+                const nodes = [];
+                for (const c of captures) {
+                  if (c.name === captureName) nodes.push(c.node.text);
+                }
+                if (nodes.length === 0) return !isPositive;
+                return nodes.every((text) => values.includes(text)) === isPositive;
+              });
+              break;
+            }
+
+            default:
+              predicates[i].push({ operator, operands: steps.slice(1) });
+          }
+
+          steps.length = 0;
+        }
+      }
+
+      Object.freeze(setProperties[i]);
+      Object.freeze(assertedProperties[i]);
+      Object.freeze(refutedProperties[i]);
```

Comment:
Another example of feedback that should've gone in the last PR, but would it make sense to freeze `predicates` here as well? I don't think it needs to remain open to later mutation.